src/main.ts

```diff
@@ -345,41 +358,73 @@ export default class ReadwisePlugin extends Plugin {
     window.clearInterval(this.scheduleInterval);
     this.scheduleInterval = null;
     if (!milliseconds) {
-      // we got manual option
+      // user set frequency to manual
       return;
     }
-    this.scheduleInterval = window.setInterval(() => this.requestArchive(null, null, true), milliseconds);
+    this.scheduleInterval = window.setInterval(() => this.refreshBookExport(undefined, true), milliseconds);
     this.registerInterval(this.scheduleInterval);
   }
 
-  refreshBookExport(bookIds?: Array<string>) {
-    bookIds = bookIds || this.settings.booksToRefresh;
-    if (!bookIds.length || !this.settings.refreshBooks) {
+  /** Initiates a sync, or queues refresh for later if queue fails.
+   * ALL syncing goes through this function. */
+  async refreshBookExport(
+    /** optional list of specific book IDs to sync */
+    bookIds?: Array<string>,
+
+    /** if true, was not initiated by user */
+    auto?: boolean,
+  ) {
+    if (!this.settings.token) return;
+
+    const targetBookIds = bookIds || this.settings.booksToRefresh;
+
+    // add potentially-missing books to booksToRefresh (TODO - prob a lil inefficient? 🤷)
+    const knownFilesPaths = Object.keys(this.settings.booksIDsMap);

```

Comment:
Do I understand correctly that the code below is why you may not need the `app.vault.on("delete")...` handler?

Reply:
Correct :heavy_check_mark: This means that the `booksIDsMap` stays more consistent as it always representing *all* known book IDs which makes re-syncing deleted files easier - just look at the delta of filesystem vs `booksIDsMap`.

Fuzzy on this, but also believe I think `on("delete")` had potential to behave in unexpected ways if you deleted many files at once (eg a whole folder) because you could not ensure that its operations finished before the next `on("delete")` fires -- so the operations within this would compete with each other when modifying the `data.json` & saving.

Reply:
Niice